### PR TITLE
Happy Thoughts

### PR DESCRIPTION
**Pull request by johannacatalinismith**
## Netlify link
https://happyythoughts.netlify.app/

### CODE REVIEW

File: src/Likes.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      body: "",
+    });
+    setLikes(likes + 1);
+  };
+
+  return (
+    <div className="likes">
+      <button
+        // clsx is to make the button pink when its been liked. clsx let us add classes when there is a special condition. https://github.com/lukeed/clsx 
+        className={clsx("likeButton", { likeButtonWithLikes: likes > 0 })}
```
Comment (schouenkes) [inline src/Likes.jsx:25]: Nice to see another solution to be able to add css classes!

File: src/Post.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { Likes } from "./Likes";
+dayjs.extend(relativeTime);
+
+export const Post = (props) => {
+  return (
+    <div className="post">
+      <div className="message">{props.message}</div>
+      <div className="metadata">
+        <Likes id={props.id} hearts={props.hearts} />{" "}
+        <div className="time">{dayjs().to(dayjs(props.createdAt))}</div>
```
Comment (schouenkes) [inline src/Post.jsx:13]: Fun to see that you have been able to simplify the code by using dayjs 😃 I will have that in mind for future projects!

**Review by schouenkes — COMMENTED** [review]
Your application follows the design and is looking good on different screen sizes. It fulfills the requirements for the assignment ✅. 

It didn't refresh the page automatically when I posted a new thought, maybe that would be something to work on if you want to do something more with it. 
For better accessibility the color contrast has to change, but otherwise it appears to follow the guidelines.

I like your naming of the components , variables, classes etc and think they are clear and understandable.

It was fun to see a different solution where you have used third party libraries to solve some of the tasks. It will be something I will have in mind for future projects.

Nice work! 🌟

**Review by dzc1 — APPROVED** [review]
Hey Joahanna! 

Amazing work, everything is looking good except 1 simple thing you left out. 

Basically, if you post a message/thought from your app, you have to perform a hard refresh on the webPage. This can be worked around using the dependency array [] on the useEffect on your `Feed.jsx` to have as a dependency the 'posts' reactive value that will trigger the fetching of new data coming from the api and thus, updating the ui. It should look something like this: 

 ```
// the useEffect hook allows us to run code once when the component is rendered
  useEffect(() => {
    // here we are fetching the data from the API, code snippet I got from the weather app project and adjusted
    fetch("https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts")
      .then((response) => response.json())
      .then((posts) => {
        setPosts(posts);
      });
  }, [posts]);
```

Now, let's make a deal, I will be giving you an approved but please be sure to update this syntax on your end to have ALL requirements met for the grade 😉. 

[Deal? ](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExYzRwejhmN29kc29meHNramtzemJhY2trYmd3anAyMmhkYTBqNHU4YyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/HOuFeZz75TBypCoS9y/giphy.gif)

Keep up the great work Johanna, React is super cool!

**Comment by johannacatalinismith** [comment]
The problem with clear msg and load comments are now fixed :)

**Pull request closed without merging**